### PR TITLE
Order completion

### PR DESCRIPTION
**Pull request by cpacia**
- [x]  Create order completion protobuf
- [x]  Refactor rating signature in `OrderConfirmation` and `OrderFulfillment` 
- [x]  Create api call to complete order
- [x]  Create, send and handle the order completion network message
- [x]  Sign payout if moderated order
- [x]  Complete moderated payout on vendor side
- [x]  Save rating on vendor side and update overall rating
- [x]  Update ratings in listing index
- [x] Create direct purchase integration test
- [x] Create moderated purchase integration test
- [x]  Update api documentation

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Overall look pretty good. Should we not have Go tests? This seems like the kind of stuff we want tested very thoroughly.

File: api/jsonapi.go [hunk 1]
```diff
@@ -1308,3 +1308,52 @@ func (i *jsonAPIHandler) POSTOrderFulfill(w http.ResponseWriter, r *http.Request
...
+		ErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	contract, state, _, records, _, err := i.node.Datastore.Purchases().GetByOrderId(or.OrderId)
+	if err != nil {
+		ErrorResponse(w, http.StatusNotFound, "order not found")
+		return
+	}
+	for _, rd := range or.Ratings {
+		if rd.Slug == "" {
```
Comment (cpacia) [inline api/jsonapi.go]: Yeah I might pull that out later when I create the code to serve it up.

File: core/completion.go [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+	contract.Signatures = append(contract.Signatures, s)
+	return contract, nil
+}
+
+func (n *OpenBazaarNode) ValidateOrderCompletion(contract *pb.RicardianContract) error {
+	if err := verifySignaturesOnOrderCompletion(contract); err != nil {
+		return err
+	}
+
+	// TODO: validate bitcoin signatures of moderated
```
Comment (cpacia) [inline core/completion.go]: Yeah I need to write the moderation code before I can do that part.

File: core/completion.go [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+
+	return nil
+}
+
+func (n *OpenBazaarNode) ValidateAndSaveRating(contract *pb.RicardianContract) error {
```
Comment (cpacia) [inline core/completion.go]: good point. I'll pull that out a little later.

**Comment by cpacia** [comment]
Yeah I'm going to try to write a bunch of tests after this order flow is done. For now there are integrations tests that are working.

**Pull request merged**